web/src/screens/NewUser.js

```diff
@@ -0,0 +1,106 @@
+// Add Imports
+import React, { useState } from "react";
+import firebase from "firebase/app";
+import "firebase/firestore";
+
+/* User creation module */
+export default function NewUser() {
+
+  const [email, setEmail] = useState("");
+  const [confirmEmail, setConfirmEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
+  const [lastError, setLastError] = useState("");
+  
+  function validateInput() {
+    if (email !== confirmEmail || password !== confirmPassword) {
+      return 0;
+    } else if (email.length <= 0 || password.length <= 0) {
+      return 0;
+    } else {
+      // courtesy email validation
+      // eslint-disable-next-line
+      const emailRegex = /(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/
+      var matches = email.match(emailRegex);
+
+      // check if the attempted registration uses a UW email
+      var isUwEmail = email.includes("@uw.edu") || email.includes("washington.edu");
```

Comment:
missing @ for `washington.edu`

Reply:
intended - this is for cases where people have @*.washington.edu. I can overhaul this and make it more robust if you want, though.